### ISSUE TITLE
GitHub repo name

### ISSUE THREAD

**Issue by muneeb-ali** [stacks-network/stacks-blockchain#3794]
Hi everyone! 

It's great to see increased open-source contributions lately (I assume Bitcoin Summer hackathon/sprint can be one reason!). I've been meaning to start a discussion topic here RE the name of this Github repo. I think it is safe to say that this is the most imp Github repo and as the decentralized ecosystem continues to grow we expect more devs and users to discover and interact with it. 

I think the name stacks-blockchain doesn't accurately represent what users should/would expect from this repo, especially after Nakamoto release. The comms WG is moving to a Bitcoin L2 terminology which will appear at various places in the ecosystem in the coming months. See [this for more details](https://github.com/stacks-network/stacks/blob/master/stacks-l2.md) on Stacks as L2 or not. Here are some points to note:

1. If most users and devs get introduced to the project as a Bitcoin L2 then seeing the term "blockchain" in the repo might be confusing. Stacks L2 will still obviously have chain state and a ledger but the term "blockchain" typically refers to L1 chains.

2. This repo might start including code for sBTC signers and even validating blocks (see latest proposal from Aaron for [better blocks](https://forum.stacks.org/t/rfc-proposal-for-collaborative-mining-with-threshold-signatures/15176?u=muneeb)). With this expanded functionality the term "blockchain" might not be appropriate e.g., if I'm running a signer or a validator, using stacks-blockchain repo for that might not be intuitive.

Given the above reasons, I want to proposal that we rename stacks-blockchain repo to **stacks-core**. This name will better fit the convention where the reference implementation of Bitcoin is called bitcoin-core. I understand that this renaming might break some links (although Github has been very good about redirecting things) and might cause some other headaches for devs and automated testing tools. So we should definitely discuss it further and see if the potential benefits are worth it and the potential downsides can be mitigated.

This is a good time to have this discussion given the rest of the Github repos and README files are also getting cleaned up and updated to bette reflect what the repos are used for now (vs some earlier state). Thanks!

**Comment by stjepangolemac** [comment]
Personally, I like the idea of renaming the repo to **stacks-core**.

I also think the name **stacks** could work, although with the GitHub organization name it might be a bit uncommon (maybe not) - **stacks/stacks**.

Some examples of main repositories with minimal names:

- https://github.com/mimblewimble/grin
- https://github.com/maticnetwork/bor
- https://github.com/ethereum-optimism/optimism
- https://github.com/OffchainLabs/nitro

**Comment by muneeb-ali** [comment]
Thanks @stjepangolemac. Yeah, the stacks-network/stacks repo can contain documentation and high-level details etc. This is similar to how other projects like IPFS have https://github.com/ipfs/ipfs for docs and then the reference implementations (they have multiple implementations).

I'd love to hear input from other people as well -- thanks!

**Comment by wileyj** [comment]
I like the idea @muneeb-ali , but you do raise some concerns, namely around redirects using the old repo name. 
We'll have to run some tests to see if that will be problematic, but you've summarized what i've thought for a while - `stacks-blockchain` just doesn't feel accurate. 

I'm also curious what would happen to forks/stars (as well as specific repo settings around permissions) if we were to rename it, but i think we can run a small test scenario to see what happens. 

As long we don't hit any major issues with those, this seems like a proposal that makes a lot of sense, with the added bonus of typing less characters.

**Comment by muneeb-ali** [comment]
> "you've summarized what i've thought for a while - stacks-blockchain just doesn't feel accurate."

Yep, agreed! 

> "I'm also curious what would happen to forks/stars"

The stars and people watching the repo will remain the same. We've done some such renaming before (even on this repo) and that aspect was fine. I think forks also auto redirect but should confirm if that is still the case. I only tested this type of thing a few years back and something might've changed.

> "with the added bonus of typing less characters."

Ha, yes! I think with a name like **stacks-core** it also immediately makes it obvious to new devs that this is like "bitcoin-core" and a reference implementation of a protocol.

So far I'm seeing support here e.g., @friedger's thumbs up above. Would love to hear from other folks 🙏

**Comment by 314159265359879** [comment]
I like "stacks-core" too, especially as it is in line with "bitcoin-core".

Additionally stacks-core is more distinct then just "stacks" which could help findability.

**Comment by postmeback** [comment]
I also like 'stacks-core'.

**Comment by wileyj** [comment]
should we move this to a discussion @muneeb-ali ?

**Comment by muneeb-ali** [comment]
@wileyj yep please to a discussion. I wasn't sure what tag to use to mark as discussion :-)
